### PR TITLE
make rnd console require science access

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

yeah how did we not have this yet.

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Unlocking research on the RND console now requires science access.

### CODE REVIEW

**Comment by Myakot** [comment]
Sorry to butt in, but maybe make it require RDs access?

**Comment by deltanedas** [comment]
without rd or cap it would be impossible to do research

File: Content.Server/Research/Systems/ResearchSystem.Console.cs [hunk 1]
```diff
@@ -18,9 +19,18 @@ private void InitializeConsole()
...
     {
+        if (args.Session.AttachedEntity is not { } ent)
+            return;
+
         if (!this.IsPowered(uid, EntityManager))
             return;
 
+        if (TryComp<AccessReaderComponent>(uid, out var access) && !_accessReader.IsAllowed(ent, access))
+        {
+            _popup.PopupEntity(Loc.GetString("research-console-no-access-popup"), ent);
```
Comment (AJCM-git) [inline Content.Server/Research/Systems/ResearchSystem.Console.cs:30]: make it gray the button instead (bonus points if tooltip)

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Client/Research/UI/ResearchConsoleMenu.xaml.cs [hunk 2]
```diff
@@ -60,10 +66,14 @@ public void  UpdatePanels(ResearchConsoleBoundInterfaceState state)
         {
             MinHeight = 10
         });
+
+        var hasAccess = _player.LocalPlayer?.ControlledEntity is not { } local ||
+                        !_entity.TryGetComponent<AccessReaderComponent>(Entity, out var access) ||
+                        _accessReader.IsAllowed(local, access);
```
Comment (AJCM-git) [inline Content.Client/Research/UI/ResearchConsoleMenu.xaml.cs:72]: I think making it a field in the state, then only doing one check in the server would be better

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Client/Research/UI/ResearchConsoleMenu.xaml.cs (hunk 2 above)
Comment (EmoGarbage404) [reply Content.Client/Research/UI/ResearchConsoleMenu.xaml.cs:72]: The only issue i had with that is that whether or not an individual has access to the server can vary across clients. The ui state updating isn't bound to a specific person looking at it so this ended up being the cleanest way to do it. Plus, it shouldn't be a huge problem considering access is predicted and this is only getting checked once.

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**